tests/test_cli_compile.py

```diff
@@ -1656,3 +1656,119 @@ def test_triple_equal_pinned_dependency_is_used(
+    (
+        pytest.param(
+            "setup.cfg",
+            """
+                [metadata]
+                name = sample_lib
+                author = Vincent Driessen
+                author_email = me@nvie.com
+
+                [options]
+                packages = find:
+                install_requires =
+                    small-fake-a==0.1
+                    small-fake-b==0.2
+
+                [options.extras_require]
+                dev =
+                    small-fake-c==0.3
+                    small-fake-d==0.4
+                test =
+                    small-fake-e==0.5
+                    small-fake-f==0.6
+            """,
+            id="setup.cfg",
+        ),
+        pytest.param(
+            "setup.py",
+            """
+                from setuptools import setup
+
+                setup(
+                    name="sample_lib",
+                    version=0.1,
+                    install_requires=["small-fake-a==0.1", "small-fake-b==0.2"],
+                    extras_require={
+                        "dev": ["small-fake-c==0.3", "small-fake-d==0.4"],
+                        "test": ["small-fake-e==0.5", "small-fake-f==0.6"],
+                    },
+                )
+            """,
+            id="setup.py",
+        ),
+        pytest.param(
+            "pyproject.toml",
+            """
+                [build-system]
+                requires = ["flit_core >=2,<4"]
+                build-backend = "flit_core.buildapi"
+
+                [tool.flit.metadata]
+                module = "sample_lib"
+                author = "Vincent Driessen"
+                author-email = "me@nvie.com"
+
+                requires = ["small-fake-a==0.1", "small-fake-b==0.2"]
+
+                [tool.flit.metadata.requires-extra]
+                dev  = ["small-fake-c==0.3", "small-fake-d==0.4"]
+                test = ["small-fake-e==0.5", "small-fake-f==0.6"]
+            """,
+            id="flit",
+        ),
+        pytest.param(
+            "pyproject.toml",
+            """
+                [build-system]
+                requires = ["poetry_core>=1.0.0"]
+                build-backend = "poetry.core.masonry.api"
+
+                [tool.poetry]
+                name = "sample_lib"
+                version = "0.1.0"
+                description = ""
+                authors = ["Vincent Driessen <me@nvie.com>"]
+
+                [tool.poetry.dependencies]
+                python = "*"
+                small-fake-a = "0.1"
+                small-fake-b = "0.2"
+
+                small-fake-c = "0.3"
+                small-fake-d = "0.4"
+                small-fake-e = "0.5"
+                small-fake-f = "0.6"
+
+                [tool.poetry.extras]
+                dev  = ["small-fake-c", "small-fake-d"]
+                test = ["small-fake-e", "small-fake-f"]
+            """,
+            id="poetry",
+        ),
+    ),
+)
+def test_input_formats(runner, tmpdir, fname, content):
```

Comment:
`tmpdir` is sorta deprecated (well, at least there's a mention of it being replaced). It's always better to use `tmp_path` which does the same except it injects `pathlib.Path` objects instead of `py.path` which are nicer to work with.

Reply:
Yep, I know. But all other tests in this file use `tmpdir`. I think it's better to keep it consistent and migrate all at once.